adapters/bidstack/bidstack.go

```diff
@@ -0,0 +1,123 @@
+package bidstack
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/mxmCherry/openrtb/v16/openrtb2"
+
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+const (
+	currencyUSDISO4217 = "USD"
+
+	headerKeyAuthorization         = "Authorization"
+	headerValueAuthorizationBearer = "Bearer "
+	headerKeyContentType           = "Content-Type"
+
+	contentTypeApplicationJSON = "application/json"
+)
+
+var (
+	ErrNoPublisherID = errors.New("publisher ID is missing")
+)
+
+type adapter struct {
+	endpoint string
+}
+
+// Builder builds a new instance of the Bidstack adapter for the given bidder with the given config.
+func Builder(_ openrtb_ext.BidderName, config config.Adapter, _ config.Server) (adapters.Bidder, error) {
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+
+	return bidder, nil
+}
+
+func (a adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	headers, err := prepareHeaders(request)
+	if err != nil {
+		return nil, []error{fmt.Errorf("headers prepare: %v", err)}
+	}
+
+	for _, imp := range request.Imp {
+		if imp.BidFloor > 0 && imp.BidFloorCur != "" && strings.ToUpper(imp.BidFloorCur) != currencyUSDISO4217 {
+			convertedValue, err := reqInfo.ConvertCurrency(imp.BidFloor, imp.BidFloorCur, currencyUSDISO4217)
+			if err != nil {
+				return nil, []error{fmt.Errorf("currency convert: %v", err)}
```

Comment:
This sample is taken from the documentation, and I believe this is a better way rather than skipping an impression because otherwise, this error case will become silent and untestable.